### ISSUE TITLE
Varifier false positive when the functional interface is inferred by the generic type

### ISSUE THREAD

**Issue by ben-manes** [google/error-prone#4578]
> /Users/ben/projects/caffeine/simulator/src/main/java/com/github/benmanes/caffeine/cache/simulator/parser/AbstractTraceReader.java:72: warning: [Varifier] Consider using > `var` here to avoid boilerplate.
> 
>       List<UnaryOperator<InputStream>> extractors = List.of(
>                                        ^
>     (see https://errorprone.info/bugpattern/Varifier)
>   Did you mean 'var extractors = List.of('?

This suggestion is not feasible because inferring to the functional interface must be done explicitly.

```java
List<UnaryOperator<InputStream>> extractors = List.of(
    in -> tryXz(in), in -> tryCompressed(in), this::tryArchived);
```

 To satisfy this check requires using a type witness. If that was intended then the suggestion should use include that.
